### PR TITLE
Adding error pages to handle HTTP errors

### PR DESCRIPTION
**Pull request by rockstardev**
Resolves: #275

As you can tell from the screenshot the _look_ of pages is vastly improved, as requested in issue.

errorpage02

### CODE REVIEW

File: BTCPayServer/Views/Error/404.cshtml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿@{
+    ViewData["ErrorTitle"] = "404 - Page not found";
+}
+
+<p class="lead-login">
+    This is like searching for person more beautiful than <a href="https://twitter.com/NicolasDorier" target="_blank">Nicolas Dorier</a>.
```
Comment (Kukks) [inline BTCPayServer/Views/Error/404.cshtml]: `for a person` *

File: BTCPayServer/Hosting/Startup.cs [hunk 2]
```diff
@@ -241,6 +241,10 @@ private static void ConfigureCore(IApplicationBuilder app, IWebHostEnvironment e
             forwardingOptions.KnownProxies.Clear();
             forwardingOptions.ForwardedHeaders = ForwardedHeaders.All;
             app.UseForwardedHeaders(forwardingOptions);
+
+
+            app.UseStatusCodePagesWithReExecute("/Error/Handle", "?statusCode={0}");
```
Comment (NicolasDorier) [inline BTCPayServer/Hosting/Startup.cs]: Can you check that the API calls are not impacted by this change? Say if you call a get invoice on the bitpay API, it should not rewrite.

Comment (rockstardev) [reply BTCPayServer/Hosting/Startup.cs]: You are handling `BitpayHttpException` in middleware so they continue to produce same responses as before.

https://github.com/btcpayserver/btcpayserver/blob/master/BTCPayServer/Hosting/BTCpayMiddleware.cs#L68

The problem is that NBitpayClient doesn't expose errors returned so we can't test for them.

Comment (rockstardev) [reply BTCPayServer/Hosting/Startup.cs]: Figured it out... it was bug in the pipeline; see: 82393eb8bbcaa10865b21e890c01f41eeb389660

**Pull request merged**